ntp-client/src/main.rs

```diff
@@ -65,33 +65,90 @@ async fn main() -> std::io::Result<()> {
 
     let exit_code = match cli.command {
         Command::Peers => {
-            let mut stream = tokio::net::UnixStream::connect(observation).await?;
-
-            let mut msg = Vec::with_capacity(16 * 1024);
-            let output: ObservableState =
-                ntp_daemon::sockets::read_json(&mut stream, &mut msg).await?;
-
-            println!("{}", serde_json::to_string_pretty(&output.peers)?);
-
-            0
+            match tokio::net::UnixStream::connect(&observation).await {
```

Comment:
Looks like a lot of code duplication.

Reply:
Refactored stream creation out of the command split. The rest is unfortunately not really simplifyable here.